### ISSUE TITLE
Use the embedded Windows 1809 OpenSSH tooling, including SSH Agent Service

### ISSUE THREAD

**Issue by joltmode** [liximomo/vscode-sftp#507]
**Is this a similar or duplicate feature request?**
- [ ] I don't know. I will go check it.
- [x] No.

**Is your feature request related to a problem? Please describe.**
The extension is currently tied to UNIX based ssh-agent solutions. [Since Windows 1809 update, Windows has a built in Service for SSH agent.](https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse) AFAIK, I cannot use the built-in SSH agent with this extension.

**Describe the solution you'd like**
To utilize the built in OpenSSH version/instances instead of custom ones.

**Describe alternatives you've considered**
None.

**Does this project help you?**
- [ ] Yes. SFTP IS AWESOME!
- [ ] Yes. [I want to support you.](https://github.com/liximomo/vscode-sftp#donation)
- [x] Not yet, because this is a fundamental problem.

**Comment by liximomo** [comment]
Try the v1.12.2.

**Comment by stefanfritsch** [comment]
Could you please elaborate on that? I tried v1.12.2 but how would the config look like?

The obvious choice

    "agent": "ssh-agent"

doesn't work.

**Comment by Eagle3386** [comment]
Of course, since the Wiki [already states](https://github.com/liximomo/vscode-sftp/wiki/SFTP-only-Config) valid values.

Running Win10, 1809 (RS5), my *working* config (shortened for brevity) looks like this:
```
{
  (...)
  "privateKeyPath":"X:\\WS\\Private.key",
  "protocol":"sftp",
  "secure":true,
  "username":"******"
  (...)
}
```

**Comment by stefanfritsch** [comment]
@Eagle3386 I'm sorry, but I really fail to see your point. I think there's a misunderstanding.

This issue is "Use the embedded Windows 1809 OpenSSH tooling, including SSH Agent Service". I talked about valid values for using the Windows ssh-agent because liximomo kinda implied that it should work in v1.12.2.

ssh-agent is not pageant and doesn't produce a socket file like cygwin's ssh-agent so I was asking what the correct setting for agent is.

**Comment by Eagle3386** [comment]
That's a pity, cause your reply seems to confirm that you failed just once more..

One last try from my end:
I'm on that very same Windows version as you and with no PuTTY whatsoever installed, just Windows's builtin OpenSSH stuff - and with my aforementioned config, it simply works.

**Comment by stefanfritsch** [comment]
1. This issue is about ssh-agent. 
2. Your solution does not use ssh-agent.

Please see [here](https://www.ssh.com/ssh/agent).

**Comment by Eagle3386** [comment]
Since you're still not reading the whole thing I wrote, but insist like a stubborn person, I give up.
Good luck, you're on your own.

**Comment by JustinGrote** [comment]
@stefanfritsch @joltmode @liximomo 

I use KeeAgent with KeePass to store and serve up my SSH agent credentials for Windows OpenSSH. I told it to create a socket and then pointed the extension to that socket and it works great. Maybe that will help.

![image](https://user-images.githubusercontent.com/15258962/75834370-a2f66b00-5d70-11ea-8be5-456a5d0a59c8.png)

![image](https://user-images.githubusercontent.com/15258962/75834391-b0abf080-5d70-11ea-9883-cf97e334f3a4.png)
